internal/ethapi/simulate.go

```diff
@@ -168,9 +168,10 @@ func (sim *simulator) processBlock(ctx context.Context, block *simBlock, header,
 			}
 		}
 	}
-	if sim.chainConfig.IsCancun(header.Number, header.Time, types.DeserializeHeaderExtraInformation(header).ArbOSFormatVersion) {
+	arbOSVersion := types.DeserializeHeaderExtraInformation(header).ArbOSFormatVersion
```

Comment:
header here is an output, not input. So you should use arbos version in parent and call it "parentArbosVersion" or something similar.
This will mean simulator will fail if it's asked to simulate a block with an arbos-upgrade and a Prague-only tx in the same block.. we can mention it in a comment or something but I don't think we care.

Reply:
Okay. This is done.

But, also! Since we don't support EIP-7685 general purpose Consensus layer requests for arbitrum, those code blocks are also explicitly disabled now in this PR.

This was causing problems in some of our tests in the `nitro` repo because there is some subtlety around getting the `RequestsHash` field populated correctly when there are no corresponding requests to handle in a post-Prague block.

Reply:
unused arbosVersion var

Reply:
It's not unused. It's used on line 173. This was the behavior of the "IsCancun" check before I started working on this PR. The outer check looked at the version from the header, and the inner check used the version from the parent.